addon/components/au-icon.gts

```diff
@@ -39,16 +40,30 @@ export default class AuIcon extends Component<AuIconSignature> {
     }
   }
 
+  get iconComponent() {
+    return typeof this.args.icon !== 'string' && this.args.icon;
+  }
+
   <template>
-    <svg
-      role="img"
-      class="au-c-icon au-c-icon--{{@icon}} {{this.alignment}} {{this.size}}"
-      aria-hidden={{this.ariaHidden}}
-      ...attributes
-    >
-      <use
-        xlink:href="{{this.iconPrefix}}@appuniversum/ember-appuniversum/appuniversum-symbolset.svg#{{@icon}}"
-      ></use>
-    </svg>
+    {{#if this.iconComponent}}
+      {{#let this.iconComponent as |Icon|}}
+        {{! @glint-expect-error: glint doesn't like us setting attributes on the passed component }}
+        <Icon
+          class="au-c-icon {{this.alignment}} {{this.size}}"
+          aria-hidden={{this.ariaHidden}}
+        />
```

Comment:
I would also spread the incoming attributes on this component. That way consumers can pass extra attributes to the AuIcon, and they would still be passed down.

```suggestion
          ...attributes
        />
```

```hbs
<AuIcon @icon={{SomeIconComponent}} class="hello" />
```

I'm not sure how that translates into the component signature types though. In theory the "Element" of the signature is then dynamic. 'SVGSVGElement' if `@icon` is a string, and `Element` otherwise, since we don't know what specific element the passed in component uses. I'm not sure if that is even possible, but I'm not very experienced in TS.

Reply:
I (as demonstrated in this PR) don't know anything about the new glint types. We could make `AuIcon` generic, taking the Element type as the type argument, but that would only be meaningful if there was a way to specify that in the `*.gts` template syntax. The `SuperTable` example in [the docs](https://typed-ember.gitbook.io/glint/environments/ember/component-signatures) actually is generic, but there's no example of it being used. We can probably stick to `Element` for now.